frameworks/AutoGluon/exec.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import os
+import warnings
+warnings.simplefilter("ignore")
+
+import matplotlib
+import numpy as np
+import pandas as pd
+matplotlib.use('agg')  # no need for tk
+
+from autogluon.task.tabular_prediction.tabular_prediction import TabularPrediction as task
+from autogluon.utils.tabular.utils.savers import save_pd, save_pkl
+import autogluon.utils.tabular.metrics as metrics
+
+from frameworks.shared.callee import call_run, result, Timer, touch
+
+log = logging.getLogger(__name__)
+
+
+def run(dataset, config):
+    log.info("\n**** AutoGluon ****\n")
+
+    metrics_mapping = dict(
+        acc=metrics.accuracy,
+        auc=metrics.roc_auc,
+        f1=metrics.f1,
+        logloss=metrics.log_loss,
+        mae=metrics.mean_absolute_error,
+        mse=metrics.mean_squared_error,
+        r2=metrics.r2
+    )
+
+    perf_metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
+    if perf_metric is None:
+        # TODO: figure out if we are going to blindly pass metrics through, or if we use a strict mapping
+        log.warning("Performance metric %s not supported.", config.metric)
+
+    is_classification = config.type == 'classification'
+
+    train = pd.DataFrame(dataset.train.data, columns=dataset.columns)
+    label = dataset.target.name
+
+    output_dir = make_subdir("models", config)
+    with Timer() as training:
+        predictor = task.fit(
+            train_data=train,
+            label=label,
+            output_directory=output_dir,
+            time_limits=config.max_runtime_seconds,
+            eval_metric=perf_metric.name,
+        )
+
+    test = pd.DataFrame(dataset.test.data, columns=dataset.columns)
+    X_test = test.drop(columns=label)
+    y_test = test[label]
+
+    with Timer() as predict:
+        predictions = predictor.predict(X_test)
+
+    probabilities = predictor.predict_proba(X_test, as_pandas=True) if is_classification else None
+    if is_classification and len(probabilities.shape) == 1:
+        # for binary, AutoGluon returns only one probability and it's not systematically the smaller or greater label.
+        # Here, we're trying to identify to which label correspond the probabilities
+        # and from there we can build the probabilities for all labels.
+        classes = dataset.target.classes
+        prob_class = next((predictions[i] if p > 0.5 else next(c for c in classes if c != predictions[i])
```

Comment:
Soon in latest AutoGluon mainline (will be in 0.0.7 release, which will likely occur by EOW), you can do this:

```
probabilities = predictor.predict_proba(dataset=X_test, as_pandas=True, as_multiclass=True) if is_classification else None
```

This should avoid the need for the entire if statement code. You will only need to check and confirm that dataset.target.classes == predictor.class_labels (if not, you may have to reorder output columns)

PR with fix: https://github.com/awslabs/autogluon/pull/447